api/v1/model_types.go

```diff
@@ -27,6 +27,8 @@ import (
 // +kubebuilder:validation:XValidation:rule="!self.url.startsWith(\"oss://\") || has(self.cacheProfile)", message="urls of format \"oss://...\" only supported when using a cacheProfile"
 // +kubebuilder:validation:XValidation:rule="!has(self.maxReplicas) || self.minReplicas <= self.maxReplicas", message="minReplicas should be less than or equal to maxReplicas."
 // +kubebuilder:validation:XValidation:rule="!has(self.adapters) || self.engine == \"VLLM\"", message="adapters only supported with VLLM engine."
+// +kubebuilder:validation:XValidation:rule="!has(self.files) || self.files.size() <= 1 || self.files.all(file, self.files.filter(other, other.path == file.path).size() == 1)", message="All file paths must be unique."
+// +kubebuilder:validation:XValidation:rule="!has(self.files) || self.files.map(f, size(f.content)).sum() <= 500000", message="A maximum of 500,000 characters are allowed across all files."
```

Comment:
is this due to k8s limits?

Reply:
Yes, I need to find a way to add comments that explain the why behind these but that wont make their way into the API reference docs.